airbyte-cdk/java/airbyte-cdk/db-sources/src/main/kotlin/io/airbyte/cdk/components/debezium/StateFilesAccessor.kt

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+package io.airbyte.cdk.components.debezium
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.google.common.collect.ImmutableList
+import io.airbyte.commons.jackson.MoreMappers
+import io.debezium.config.Configuration
+import io.debezium.relational.history.AbstractFileBasedSchemaHistory
+import io.debezium.relational.history.HistoryRecord
+import io.debezium.relational.history.HistoryRecordComparator
+import io.debezium.relational.history.SchemaHistoryListener
+import io.debezium.storage.file.history.FileSchemaHistory
+import java.io.IOException
+import java.io.UncheckedIOException
+import java.lang.reflect.InvocationTargetException
+import java.lang.reflect.Method
+import java.nio.ByteBuffer
+import java.nio.file.Files
+import java.nio.file.Path
+import java.util.function.Consumer
+import kotlin.Exception
+import kotlin.RuntimeException
+import kotlin.String
+import org.apache.commons.io.FileUtils
+import org.apache.kafka.common.utils.ByteBufferInputStream
+import org.apache.kafka.connect.json.JsonConverter
+import org.apache.kafka.connect.json.JsonConverterConfig
+import org.apache.kafka.connect.runtime.WorkerConfig
+import org.apache.kafka.connect.runtime.standalone.StandaloneConfig
+import org.apache.kafka.connect.storage.Converter
+import org.apache.kafka.connect.storage.FileOffsetBackingStore
+
+/**
+ * [StateFilesAccessor] reads and writes from Debezium offset and schema history files. This should
+ * NOT be done while the Debezium Engine is running! None of this is thread-safe. Instead, this
+ * object should be used to:
+ * 1. create and populate the files prior to running the Debezium Engine,
+ * 2. read the changed files after running the Debezium Engine.
+ *
+ * The files are deleted on [close].
```

Comment:
So would this mean that every iteration of the `collect` function starts and closes the engine?

Reply:
Yes, that's correct. I'm assuming that this takes a couple of seconds at most. I hope I'm not wrong!